pkg/router/template/util/haproxy/map_entry.go

```diff
@@ -58,13 +58,19 @@ func generateEdgeReencryptMapEntry(cfg *BackendConfig) *HAProxyMapEntry {
 
 // generateHttpRedirectMapEntry generates a map entry for redirecting insecure/http hosts.
 func generateHttpRedirectMapEntry(cfg *BackendConfig) *HAProxyMapEntry {
-	if len(cfg.Host) > 0 && cfg.InsecurePolicy == routev1.InsecureEdgeTerminationPolicyRedirect {
-		return &HAProxyMapEntry{
-			Key:   templateutil.GenerateRouteRegexp(cfg.Host, cfg.Path, cfg.IsWildcard),
-			Value: cfg.Name,
+	if len(cfg.Host) > 0 {
+		haproxyMapEntry := &HAProxyMapEntry{
+			Key: templateutil.GenerateRouteRegexp(cfg.Host, cfg.Path, cfg.IsWildcard),
+		}
+		switch cfg.InsecurePolicy {
+		case routev1.InsecureEdgeTerminationPolicyRedirect:
+			haproxyMapEntry.Value = "1"
+			return haproxyMapEntry
+		default:
+			haproxyMapEntry.Value = "0"
+			return haproxyMapEntry
 		}
```

Comment:
This can be simplified:
```suggestion
			Key:   templateutil.GenerateRouteRegexp(cfg.Host, cfg.Path, cfg.IsWildcard),
			Value: "0",
		}
		switch cfg.InsecurePolicy {
		case routev1.InsecureEdgeTerminationPolicyRedirect:
			haproxyMapEntry.Value = "1"
		}
		return haproxyMapEntry
```